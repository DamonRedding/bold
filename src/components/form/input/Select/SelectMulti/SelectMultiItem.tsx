import { Interpolation } from 'emotion'
import * as React from 'react'

import { Styles, withStyles, WithStylesProps } from '../../../../../styles'
import Times from '../../../../elements/Icon/generated/Times'

export interface SelectMultiItemProps extends WithStylesProps {
    style?: Interpolation
    disabled?: boolean
    onRemove(e: React.MouseEvent<HTMLSpanElement>): void
}

@withStyles
export class SelectMultiItem extends React.Component<SelectMultiItemProps> {
    render() {
        const { theme, css, style, children, onRemove, disabled, ...rest } = this.props
        const styles: Styles = {
            root: {
                border: `1px solid ${theme.pallete.divider}`,
                borderRadius: theme.radius.button,
                display: 'inline-flex',
                alignItems: 'center',
                fontWeight: 'bold',
            },
            text: {
                padding: disabled ? 'calc(0.25rem - 1px) 0.25rem' : 'calc(0.125rem - 1px) 0.25rem',
            },
            button: {
                background: theme.pallete.surface.background,
                cursor: 'pointer',
                fontSize: '1.25rem',
                display: 'inline-flex',
                alignItems: 'center',
                padding: 'calc(0.125rem - 1px) 0',
                '&:hover': {
                    color: theme.pallete.status.danger.main,
                },
                'svg': {
                    fill: 'currentColor',
                },
            },
        }
        return (
            <span className={css(styles.root, style)} {...rest}>
                <span className={css(styles.text)}>
                    {children}
                </span>
                {!disabled &&
                    <span className={css(styles.button)} onClick={onRemove} title='Remover'>
                        <Times />
                    </span>
                }
            </span>
        )
    }
}
