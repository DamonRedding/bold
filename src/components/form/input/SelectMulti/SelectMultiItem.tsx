import { Interpolation } from 'emotion'
import * as React from 'react'

import { Styles, withStyles, WithStylesProps } from '../../../../styles'
import { Times } from '../../../elements/Icon/generated/Times'

export interface SelectMultiItemProps extends WithStylesProps {
    style?: Interpolation
    onRemove(e: React.MouseEvent<HTMLSpanElement>): void
}

@withStyles
export class SelectMultiItem extends React.Component<SelectMultiItemProps> {
    render() {
        const { theme, css, style, children, onRemove, ...rest } = this.props
        const styles: Styles = {
            root: {
                border: `1px solid ${theme.pallete.divider}`,
                borderRadius: theme.radius.button,
                display: 'inline-flex',
                alignItems: 'center',
                fontWeight: 'bold',
            },
            text: {
                padding: 'calc(0.125rem - 1px) 0.25rem',
            },
            button: {
                background: theme.pallete.surface.background,
                cursor: 'pointer',
                fontSize: '1.25rem',
                display: 'inline-flex',
                alignItems: 'center',
                padding: 'calc(0.125rem - 1px) 0',
                '&:hover': {
                    color: theme.pallete.status.danger.main,
                },
                'svg': {
                    fill: 'currentColor',
                },
            },
        }
        return (
            <span className={css(styles.root, style)} {...rest}>
                <span className={css(styles.text)}>
                    {children}
                </span>
                <span className={css(styles.button)} onClick={onRemove} title='Remover'>
                    <Times />
                </span>
            </span>
        )
    }
}
