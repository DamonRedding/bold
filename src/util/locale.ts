/**
 * Return the user locale based on browser's language.
 *
 * @returns The user locale.
 */
export const getUserLocale = () => {
  if (!navigator) {
    return null
  }

  return navigator.languages && navigator.languages.length ? navigator.languages[0] : navigator.language
}
