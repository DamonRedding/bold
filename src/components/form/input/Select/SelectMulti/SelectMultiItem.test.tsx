import React from 'react'
import { fireEvent, render } from 'react-testing-library'

import { LocaleContext } from '../../../../../locale'
import ptBr from '../../../../../locale/locales/pt-BR'

import { SelectMultiItem } from './SelectMultiItem'

it('should render correctly', () => {
  const { container } = render(<SelectMultiItem style={{ color: 'red' }} onRemove={jest.fn()} />)
  expect(container).toMatchSnapshot()
})

it('should call onRemove when clicked on remove button', () => {
  const remove = jest.fn()
  const { getByTitle } = render(<SelectMultiItem onRemove={remove} />)
  expect(remove).not.toHaveBeenCalled()
  fireEvent.click(getByTitle('Remove'))
  expect(remove).toHaveBeenCalled()
})

it('should not show remove button when disabled', () => {
  const { queryByTitle } = render(<SelectMultiItem onRemove={jest.fn()} disabled />)
  expect(queryByTitle('Remove')).toBeFalsy()
})

it('should allow message customization via locale context', () => {
  const { queryByTitle } = render(
    <LocaleContext.Provider value={ptBr}>
      <SelectMultiItem onRemove={jest.fn()} disabled>
        Test
      </SelectMultiItem>
    </LocaleContext.Provider>
  )
  expect(queryByTitle('Remover')).toBeFalsy()
})
