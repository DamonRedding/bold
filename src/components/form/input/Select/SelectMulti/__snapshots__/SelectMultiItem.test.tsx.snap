// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.emotion-2 {
  border: 1px solid #DCDCE0;
  border-radius: 4px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  color: red;
}

.emotion-0 {
  padding: calc(0.125rem - 1px) 0.25rem;
}

.emotion-1 {
  background: #F2F2F7;
  cursor: pointer;
  font-size: 1.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: calc(0.125rem - 1px) 0;
}

.emotion-1:hover {
  color: #D83A2E;
}

.emotion-1 svg {
  fill: currentColor;
}

<div>
  <span
    class="emotion-2"
  >
    <span
      class="emotion-0"
    />
    <span
      class="emotion-1"
      title="Remover"
    >
      <svg
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M18.36 7.05l-1.41-1.41L12 10.59 7.05 5.64 5.64 7.05 10.59 12l-4.95 4.95 1.41 1.41L12 13.41l4.95 4.95 1.41-1.41L13.41 12l4.95-4.95z"
          data-name="times"
        />
      </svg>
    </span>
  </span>
</div>
`;
